astropy/table/table.py

```diff
@@ -799,18 +803,21 @@ def __init__(self, data=None, masked=False, names=None, dtype=None,
         elif data is None:
             if names is None:
                 if dtype is None:
-                    if meta is not None:
-                        self.meta = deepcopy(meta) if copy else meta
-                    return
-                try:
-                    # No data nor names but dtype is available.  This must be
-                    # valid to initialize a structured array.
-                    dtype = np.dtype(dtype)
-                    names = dtype.names
-                    dtype = [dtype[name] for name in names]
-                except Exception:
-                    raise ValueError('dtype was specified but could not be '
-                                     'parsed for column names')
+                    # Table was initialized as `t = Table()`. Set up for empty
+                    # table with names=[], data=[], and n_cols=0.
+                    # self._init_from_list() will simply return, giving the
+                    # expected empty table.
+                    names = []
+                else:
+                    try:
+                        # No data nor names but dtype is available.  This must be
+                        # valid to initialize a structured array.
+                        dtype = np.dtype(dtype)
+                        names = dtype.names
+                        dtype = [dtype[name] for name in names]
+                    except Exception:
+                        raise ValueError('dtype was specified but could not be '
```

Comment:
Probably good to write a few tests that cover unparsable dtype and one that does not have fields.